src/main/java/com/google/sps/servlets/RecipeServlet.java

```diff
@@ -14,22 +14,69 @@
 
 package com.google.sps.servlets;
 
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
 import com.google.gson.Gson;
+import com.google.sps.data.DatabaseReferences;
+import com.google.sps.data.Post;
+import java.io.BufferedReader;
 import java.io.IOException;
+import java.lang.StringBuilder;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 @WebServlet("/api/post")
 public class RecipeServlet extends HttpServlet {
+  private Gson gson;
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+  public void init() {
+    gson = new Gson();
+  }
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatabaseReferences.ROOT.child("post-metadata/0").addListenerForSingleValueEvent(new ValueEventListener() {
+      @Override
+      public void onDataChange(DataSnapshot dataSnapshot) {
+        String json = gson.toJson(dataSnapshot.getValue());
```

Comment:
This is where the async magic happens, need to somehow suspend execution until 'onDataChange' executes.